src/sources/aws_s3/sqs.rs

```diff
@@ -558,9 +555,10 @@ impl IngestorProcess {
                     BatchStatus::Delivered => Ok(()),
                     BatchStatus::Errored => Err(ProcessingError::ErrorAcknowledgement),
                     BatchStatus::Rejected => {
+                        // We should emit `component_discarded_events_total` here, but is count always 1?
```

Comment:
Big question here 💭

Reply:
I believe you can get the count of items (events) from the stream like this:

```
let (count, _) = stream.size_hint(); 
```

Should that also be used above in `StreamClosedError` ?

Reply:
The `StreamClosedError` _seems_ like it's dealing with a single event at a time, unless the naming is bonkers.

Reply:
How does event array counts play into this?

Reply:
Yeah, that's a big part of my uncertainty 😅

Reply:
Based on discussion with @jszwedko and @neuronull, removed this in https://github.com/vectordotdev/vector/pull/14344/commits/19c050683ec00f139b794ace2041800521a49496